Shared/Samples/Trace utility network/TraceUtilityNetworkView.Model.swift

```diff
@@ -0,0 +1,353 @@
+        ///
+        /// For junction features with more than one terminal, the user should be prompted to pick a
+        /// terminal. For edge features, the fractional point along the feature's edge should be
+        /// computed.
+        /// - Parameters:
+        ///   - feature: The feature to be added to the pending trace.
+        ///   - mapPoint: The location on the map where the feature was discovered. If the feature is a
+        ///   junction type, the feature's geometry will be used instead.
+        func add(_ feature: ArcGISFeature, at mapPoint: Point) {
+            if let element = network?.makeElement(arcGISFeature: feature),
+               let geometry = feature.geometry,
+               let table = feature.table as? ArcGISFeatureTable,
+               let networkSource = network?.definition?.networkSource(named: table.tableName) {
+                switch networkSource.kind {
+                case .junction:
+                    add(element, at: geometry)
+                    if element.assetType.terminalConfiguration?.terminals.count ?? .zero > 1 {
+                        terminalSelectorIsOpen.toggle()
+                    }
+                case .edge:
+                    if let line = GeometryEngine.makeGeometry(from: geometry, z: nil) as? Polyline {
+                        element.fractionAlongEdge = GeometryEngine.polyline(
+                            line,
+                            fractionalLengthClosestTo: mapPoint,
+                            tolerance: -1
+                        )
+                        Task {
+                            await updateUserHint(
+                                withMessage: String(format: "fractionAlongEdge: %.3f", element.fractionAlongEdge)
+                            )
+                        }
+                        add(element, at: mapPoint)
+                    }
+                @unknown default:
+                    return
+                }
+            }
+        }
+        
+        /// Identifies the first discoverable feature at the provided screen point.
+        /// - Parameters:
+        ///   - screenPoint: The location on the screen where the identify operation is desired.
+        ///   - proxy: The map view proxy to perform the identify operation with.
+        /// - Returns: The first discoverable feature or `nil` if none were identified.
+        func identifyFeatureAt(_ screenPoint: CGPoint, with proxy: MapViewProxy) async -> ArcGISFeature? {
+            guard let feature = try? await proxy.identifyLayers(
+                screenPoint: screenPoint,
+                tolerance: 10
+            ).first?.geoElements.first as? ArcGISFeature else {
+                return nil
+            }
+            return feature
+        }
+        
+        /// Makes new utility trace parameters with the provided trace type.
+        /// - Parameter type: The trace type.
+        func makeTraceParameters(with type: UtilityTraceParameters.TraceType) {
```

Comment:
This method name is pretty confusing because it starts with "make" like it is returning some trace parameters so maybe it would be good to rename or just return the `UtilityTraceParameters`.

Reply:
Since this method is called from the UI and not the model itself, I found that it didn't really make sense to return the params so I've instead opted for a rename to `initializeTraceParameters`, keeping the existing behavior and adjusting the doc to match.

Reply:
Hmm ok, I'm weary of taking transferring too much responsibility from the model to a button in the view (this would also involve opening up access to the model's `mediumVoltageRadial` property).

I'd like to strike the right balance here. Can you think of any names or other approaches that might make sense?

Reply:
Would something like `prepareForTrace(ofType:)` work for you? 

Then we could also move `model.tracingActivity = .settingPoints(pointType: .start)` from the button into that method.

Reply:
What about `setTraceParameters(of:)` ?

Reply:
I'll take it!